infra/Pos/Network/Windows/DnsDomains.hs

```diff
@@ -0,0 +1,23 @@
+{-# LANGUAGE CPP                      #-}
+{-# LANGUAGE ForeignFunctionInterface #-}
+
+#if defined(mingw32_HOST_OS)
+#define WINDOWS
+#endif
+
+module Pos.Network.Windows.DnsDomains (
+#ifdef WINDOWS
+    getWindowsDefaultDnsServer
+#endif
+    ) where
+
+#ifdef WINDOWS
+import           Universum
+
+import           Foreign.C            (CString, peekCString)
+
+foreign import ccall "getWindowsDefDnsServer" getWindowsDefDnsServer :: IO CString
+
+getWindowsDefaultDnsServer :: IO String
+getWindowsDefaultDnsServer = getWindowsDefDnsServer >>= peekCString
```

Comment:
No, if the C code uses `malloc` and returns the string then the C String would have to be explicitly deallocated on the Haskell side using `free`, or we'd have to switch things around and make the caller responsible for the string allocation in the first place.

http://hackage.haskell.org/package/base-4.10.0.0/docs/Foreign-Marshal-Alloc.html#v:free